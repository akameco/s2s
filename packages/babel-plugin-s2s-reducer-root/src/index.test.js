// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const cwd = path.resolve(__dirname, '..')
const filename = path.resolve(cwd, '__fixtures__', 'app', 'reducer.js')
const output = path.resolve(cwd, 'reducer.js')

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename, configFile: false },
  pluginOptions: {
    input: '__fixtures__/**/reducer.js',
    output,
    globOptions: { cwd },
  },
  tests: [
    {
      title: 'options',
      code: `// @flow`,
    },
    {
      title: 'no parse',
      code: `// @flow
      const state = { ...state, a: 1}
      `,
    },
  ],
})

pluginTester({
  plugin,
  babelOptions: { filename, configFile: false },
  snapshot: true,
  pluginOptions: {
    input: './**/reducer.js',
    output,
    globOptions: { cwd },
  },
  tests: [
    {
      title: 'glob options',
      code: `// show options`,
    },
  ],
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename, configFile: false },
  pluginOptions: {
    input: '__fixtures__/**/reducer.js',
    output,
    combineReducers: './myCombineReducers',
    globOptions: { cwd },
  },
  tests: [
    {
      title: 'reducer options',
      code: `// @flow`,
    },
  ],
})

pluginTester({
  plugin,
  babelOptions: { filename, configFile: false },
  tests: [
    {
      title: 'error',
      code: `// throw error`,
      error: /require input option/,
    },
  ],
})

pluginTester({
  plugin,
  babelOptions: { filename, configFile: false },
  pluginOptions: { input: '__fixtures__/**/*.js', globOptions: { cwd } },
  tests: [
    {
      title: 'error',
      code: `// throw error`,
      error: /require output option/,
    },
  ],
})
