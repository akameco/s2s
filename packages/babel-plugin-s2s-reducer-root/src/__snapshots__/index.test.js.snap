// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`s2s-reducer-root glob options: glob options 1`] = `
"
// show options

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
// show options
import { combineReducers } from \\"redux\\";
import App from \\"./__fixtures__/app/reducer\\";
import Bob from \\"./__fixtures__/bob/reducer\\";

export default combineReducers({
  App,
  Bob
});
"
`;

exports[`s2s-reducer-root no parse: no parse 1`] = `
"
// @flow
      const state = { ...state, a: 1}

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import { combineReducers } from \\"redux\\";
import App from \\"./__fixtures__/app/reducer\\";
import Bob from \\"./__fixtures__/bob/reducer\\";

export default combineReducers({
      App,
      Bob
});
"
`;

exports[`s2s-reducer-root options: options 1`] = `
"
// @flow

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import { combineReducers } from \\"redux\\";
import App from \\"./__fixtures__/app/reducer\\";
import Bob from \\"./__fixtures__/bob/reducer\\";

export default combineReducers({
  App,
  Bob
});
"
`;

exports[`s2s-reducer-root reducer options: reducer options 1`] = `
"
// @flow

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import { combineReducers } from \\"./myCombineReducers\\";
import App from \\"./__fixtures__/app/reducer\\";
import Bob from \\"./__fixtures__/bob/reducer\\";

export default combineReducers({
  App,
  Bob
});
"
`;
